invariant/examples/sql_injection/sql_injection_query_pipeline.ipynb

```diff
@@ -0,0 +1,856 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "c55d23d0-150b-4ad9-b6ff-77a12dc8767f",
+   "metadata": {},
+   "source": [
+    "# Demonstration of SQL injection by LLM using query pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "911fcc02-c66e-4d74-8281-aa24fe3f672a",
+   "metadata": {},
+   "source": [
+    "## 1. Replicate the SQL injection by a simple SQL agent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d77dc9cf-d776-41f7-97da-1863bc4ca6d1",
+   "metadata": {},
+   "source": [
+    "### Download and extract the tutorial sample database"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "id": "c636f556-7d79-409b-a2b0-714bf38b90f4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "100  298k  100  298k    0     0   311k      0 --:--:-- --:--:-- --:--:--  311k\n",
+      "curl: (6) Could not resolve host: .\n",
+      "Archive:  ./chinook.zip\n",
+      "  inflating: chinook.db              \n"
+     ]
+    }
+   ],
+   "source": [
+    "!rm -f chinook.db\n",
+    "!curl \"https://www.sqlitetutorial.net/wp-content/uploads/2018/03/chinook.zip\" -O ./chinook.zip\n",
+    "!unzip ./chinook.zip"
```

Comment:
we should also install llama-index and sqlalchemy in the beginning, sth like:
```
!pip install llama_index
!pip install sqlalchemy
```

Reply:
Thanks! Fixed.